src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
         opcode: Opcode,
     ) -> ParseResult<'a, EncodedAnnotations> {
         let input_after_opcode = self.consume(1);
-        // TODO: This implementation actively reads the annotations, which isn't necessary.
-        //       At this phase of parsing we can just identify the buffer slice that contains
-        //       the annotations and remember their encoding; later on, the annotations iterator
-        //       can actually do the reading. That optimization would be impactful for FlexSyms
-        //       that represent inline text.
```

Comment:
🗺️ I did this optimization for skipping `FlexUInt` symbol addresses, but doing it for `FlexSym` would be much more involved. I introduced two methods below to encapsulate that logic:
* `consume_flex_uint` (skips without reading)
* `consume_flex_sym` (skips by reading)